nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js

```diff
@@ -1082,6 +1116,179 @@ export function ApnResourcesEdit(props: Props) {
           ))}
         </List>
       </DialogContent>
+      <DialogActions>
+        <Button onClick={props.onClose} skin="regular">
+          Cancel
+        </Button>
+        <Button onClick={onSave} variant="contained" color="primary">
+          {props.isAdd ? 'Save And Continue' : 'Save'}
+        </Button>
+      </DialogActions>
+    </>
+  );
+}
+
+export function HeaderEnrichmentConfig(props: Props) {
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const [error, setError] = useState('');
+  const ctx = useContext(GatewayContext);
+  const [heConfig, setHeConfig] = useState<gateway_he_config>(
+    props.gateway?.cellular.he_config || DEFAULT_HE_CONFIG,
+  );
+
+  const handleHEChange = (key: string, val) => {
+    setHeConfig({...heConfig, [key]: val});
+  };
+
+  const [
+    cellularConfig,
+    _setCellularConfig,
+  ] = useState<gateway_cellular_configs>(props.gateway?.cellular || {});
+  const heEncodingTypes = ['BASE64', 'HEX2BIN'];
+  const heEncryptionAlgorithmTypes = [
+    'RC4',
+    'AES256_CBC_HMAC_MD5',
+    'AES256_ECB_HMAC_MD5',
+    'GZIPPED_AES256_ECB_SHA1',
+  ];
+  const heHashFunctionTypes = ['MD5', 'HEX', 'SHA256'];
+
+  const onSave = async () => {
+    try {
+      const gateway = {
+        ...(props.gateway || DEFAULT_GATEWAY_CONFIG),
+        cellular: {
+          ...cellularConfig,
```

Comment:
Why are we creating cellularConfig state instead of directly using props.cellularConfig here.

Reply:
I created cellularConfig because I had flow errors when I used `props.gateway.cellular`, `gateway` is a conditional  prop but it should not be except for gateway Info

Reply:
What I mean is we only create gateways in gateway info, then we always edit an existing gateway

Reply:
You can do this ...(props.gateway.cellular ?? {})